spark/src/main/scala/geotrellis/spark/io/LayerReader.scala

```diff
@@ -59,6 +62,33 @@ trait LayerReader[ID] {
 }
 
 object LayerReader {
+  /**
+   * Produce LayerReader instance based on URI description.
+   * Find instances of [[LayerReaderProvider]] through Java SPI.
+   */
+  def apply(attributeStore: AttributeStore, layerReaderUri: URI)(implicit sc: SparkContext): LayerReader[LayerId] = {
```

Comment:
@lossyrob Calling that that this `apply` is placed here even through it is producing `FilteringLayerReader` to keep the user facing API symmetrical `LayerReader`/`LayerWriter`